src/xpub.ts

```diff
@@ -305,10 +305,10 @@ class Xpub extends EventEmitter {
     const dustAmount = utils.computeDustAmount(this.crypto, txSize);
     // Abandon the change output if change output amount is less than dust amount
     if (needChangeoutput && total.minus(params.amount).minus(fee) > dustAmount) {
-      outputs.push({
-        script: this.crypto.toOutputScript(params.changeAddress),
+      outputs.unshift({
```

Comment:
I guess this is a perf optim to avoid using `filter` when checking the outputs, right?
Unfortunately it causes a minor issue where the 1st output displayed on the Nano is labeled `"Output #2"` because of this I think.
Also this is assuming that outputs are specifically coming from buildTx AND they were not modified by the caller, which is fine for us but maybe not perfectly safe. And in the end the trick is very hard to understand by just reading the code.

Reply:
With the output in last position the display issue is solved :+1: 
However consider my two other points. Do you think there's an issue using `filter` instead?